operator/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakIngressTest.java

```diff
@@ -188,4 +193,87 @@ public void testMainIngressDurability() {
                     assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(0);
                 });
     }
+
+    @Test
+    public void testCustomIngressDeletion() {
+
+        Keycloak defaultKeycloakDeployment;
+        Ingress customIngressCreatedManually;
+        String kcDeploymentName = "example-kc";
+
+        try {
+
+            Awaitility.await().atMost(1, MINUTES).untilAsserted(() -> {
+                assertThat(k8sclient.apps().statefulSets().inNamespace(namespace).withName(kcDeploymentName).get()).isNull();
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(0);
+            });
+
+            customIngressCreatedManually = createCustomIngress(kcDeploymentName, namespace, 8443);
+
+            final Resource<Ingress> customIngressDeployedManuallySelector = k8sclient
+                    .network()
+                    .v1()
+                    .ingresses()
+                    .inNamespace(namespace)
+                    .withName(customIngressCreatedManually.getMetadata().getName());
+
+            Awaitility.await().atMost(1, MINUTES).untilAsserted(() -> {
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(1);
+            });
+
+            // Delete Keycloak CR
+            Log.info("Deleting Keycloak CR and watching cleanup");
+            defaultKeycloakDeployment = K8sUtils.getDefaultKeycloakDeployment();
+            defaultKeycloakDeployment.getSpec().setIngressSpec(new IngressSpec());
+            defaultKeycloakDeployment.getSpec().getIngressSpec().setIngressEnabled(false);
+
+            K8sUtils.deployKeycloak(k8sclient, defaultKeycloakDeployment, true);
+            K8sUtils.waitForKeycloakToBeReady(k8sclient, defaultKeycloakDeployment); // wait for reconciler to calm down to avoid race condititon
+
+            Awaitility.await().untilAsserted(() -> {
+                Log.info("Make sure the Custom Ingress still remains");
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(1);
+            });
+
+            Log.info("Destroying the Custom Ingress created manually to avoid errors in others Tests methods");
+            assertThat(customIngressDeployedManuallySelector.delete()).isTrue();
+
+            Awaitility.await().untilAsserted(() -> {
+                assertThat(k8sclient.network().v1().ingresses().inNamespace(namespace).list().getItems().size()).isEqualTo(0);
+            });
```

Comment:
This should be in `finally` block in case something fails in this test.

Reply:
OK. Moved to ```finally``` block.

Done!

Thanks!